src/services/todoist.service.ts

```diff
@@ -0,0 +1,122 @@
+import { HttpService } from '@nestjs/axios'
+import { Injectable } from '@nestjs/common'
+import { lastValueFrom } from 'rxjs'
+
+import type { Task } from '../types'
+
+const LIMIT = 100
+
+type SyncDue = {
+    date: string
+    is_recurring: boolean
+    lang: string
+    string: string
+    timezone?: string | null
+}
+
+type SyncTask = {
+    added_by_uid: string
+    assigned_by_uid?: string
+    checked: boolean
+    child_order: number
+    collapsed: boolean
+    content: string
+    added_at: string
+    completed_at: string
+    description: string
+    id: string
+    is_deleted: boolean
+    labels: string[]
+    parent_id?: string | null
+    priority: number
+    project_id: string
+    responsible_uid?: string | null
+    section_id: string
+    user_id: string
+    due?: SyncDue | null
+}
+
+@Injectable()
+export class TodoistService {
+    constructor(private readonly httpService: HttpService) {}
+
+    async getCompletedTasks({
+        projectId,
+        token,
+    }: {
+        token: string
+        projectId: string
+    }): Promise<Task[]> {
+        const completedTasks = await this.getCompletedTasksInternal({ token, offset: 0, projectId })
+
+        return completedTasks.map((task) => this.getTaskFromQuickAddResponse(task))
+    }
+
+    private async getCompletedTasksInternal({
+        offset,
+        projectId,
+        token,
+    }: {
+        token: string
+        offset: number
+        projectId: string
+    }): Promise<SyncTask[]> {
+        const response = await lastValueFrom(
+            this.httpService.get<SyncTask[]>(
+                `https://api.todoist.com/sync/v9/items/get_completed?project_id=${projectId}&offset=${offset}&limit=${LIMIT}`,
```

Comment:
@scottlovegrove since this is a public project, I suggest adding a comment here that this endpoint is not public (at the time of writing, maybe adding a date to the comment would be useful), and there's no guarantees of stability (at least not the same we have for the endpoints available publicly).

Hopefully we will have some agreement on the official way of doing this soon™